src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,243 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.DataCorrectnessException;
+import core.basesyntax.exceptions.LoginDuplicateException;
+import core.basesyntax.exceptions.NullUserDataException;
+import core.basesyntax.exceptions.NullUserException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @BeforeEach
+    void setUp() {
+        User user1 = new User();
+        user1.setId(1L);
+        user1.setLogin("LordFar");
+        user1.setPassword("34gsge&ws");
+        user1.setAge(23);
```

Comment:
better create a user field to be reused across test methods
You only need to save one user in DB in the test about duplicate login - so let's save it there